client/src/main/scala/com.gu.contentapi.client/ContentApiClient.scala

```diff
@@ -1,97 +1,54 @@
 package com.gu.contentapi.client
 
-import java.io.IOException
-import java.util.concurrent.TimeUnit
 import com.gu.contentapi.client.model._
 import com.gu.contentapi.client.model.v1._
-import com.gu.contentapi.client.utils.QueryStringParams
-import com.gu.contentapi.buildinfo.CapiBuildInfo
-import okhttp3._
-import scala.concurrent.{ExecutionContext, Future, Promise}
-import scala.util.Try
 import com.gu.contentapi.client.thrift.ThriftDeserializer
-
-case class GuardianContentApiError(httpStatus: Int, httpMessage: String, errorResponse: Option[ErrorResponse] = None) extends Exception(httpMessage)
-
-trait ContentApiClientLogic {
-  val apiKey: String
-
-  protected val userAgent = "content-api-scala-client/"+CapiBuildInfo.version
-
-  protected lazy val http = new OkHttpClient.Builder()
```

Comment:
To point out possible alternatives, we can abstract away the HTTP client implementation without having to refactor the whole trait. For example, we can require a [gigahorse.HttpClient](http://eed3si9n.com/gigahorse/concepts.html#HttpClient) and let callers define which implementation to use (e.g. OkHttp, Akka HTTP, etc.) as well as all the necessary configuration.

Alternatively, we abstract away what we need from HTTP client into a trait. We can probably make do with a single abstract method: `def get(uri: URI, headers: Map[String, String]): Future[...]`.

Reply:
I did this to remove the dependency, so I'd rather go down the abstract road.

Reply:
... but still want to refactor the whole trait, as none of the methods I removed were part of the API 🤗